contracts/modules/cover/Cover.sol

```diff
@@ -0,0 +1,575 @@
+
+import "@openzeppelin/contracts-v4/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts-v4/token/ERC20/IERC20.sol";
+import "../../interfaces/ICover.sol";
+import "../../interfaces/IStakingPool.sol";
+import "../../interfaces/IPool.sol";
+import "../../abstract/MasterAwareV2.sol";
+import "../../interfaces/IMemberRoles.sol";
+import "../../interfaces/ICoverNFT.sol";
+
+
+contract Cover is ICover, MasterAwareV2 {
+
+  uint public capacityFactor;
+  uint public coverCount;
+
+  Product[] public override products;
+  ProductType[] public override productTypes;
+
+  mapping(uint => CoverData) public override covers;
+  mapping(uint => CoverChunk[]) public coverChunksForCover;
+
+  mapping(uint => uint) initialPrices;
+
+  mapping(uint => uint96) public override activeCoverAmountInNXM;
+
+  ICoverNFT public override coverNFT;
+
+
+  struct LastPrice {
+    uint96 value;
+    uint32 lastUpdateTime;
+  }
+
+  /*
+    (productId, poolAddress) => lastPrice
+    Last base prices at which a cover was sold by a pool for a particular product.
+  */
+  mapping(uint => mapping(address => LastPrice)) lastPrices;
+
+
+  /* === CONSTANTS ==== */
+
+  uint public REWARD_BPS = 5000;
+  uint public constant PERCENTAGE_CHANGE_PER_DAY_BPS = 100;
+  uint public constant BASIS_PRECISION = 10000;
+  uint public constant STAKE_SPEED_UNIT = 100000e18;
+  uint public constant PRICE_CURVE_EXPONENT = 7;
+  uint public constant MAX_PRICE_PERCENTAGE = 1e20;
+
+
+  constructor(ICoverNFT _coverNFT) {
+    coverNFT = _coverNFT;
+  }
+
+  /* === MUTATIVE FUNCTIONS ==== */
+
+  function buyCover(
+    address owner,
+    uint24 productId,
+    uint8 payoutAsset,
+    uint96 amount,
+    uint32 period,
+    uint maxPremiumInAsset,
+    CoverChunkRequest[] memory coverChunkRequests
+  ) external payable override onlyMember returns (uint /*coverId*/) {
+    require(initialPrices[productId] != 0, "Cover: product not initialized");
+
+    uint amountLeftToCoverInNXM;
+    uint tokenPrice;
+    {
+      IPool pool = pool();
+      // convert to NXM amount
+      tokenPrice = pool.getTokenPrice(payoutAsset);
```

Comment:
pool.getTokenPrice -> pool().getTokenPrice
This means you can remove the lone above.